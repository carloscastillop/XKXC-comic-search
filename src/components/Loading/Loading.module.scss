//colors
$color_mine_shaft_approx: #333;

//@extend-elements
//original selectors
//.doubleBounce1, .doubleBounce2
%extend_1 {
  width: 100%;
  height: 100%;
  //Instead of the line below you could use @include border-radius($radius, $vertical-radius)
  border-radius: 50%;
  background-color: $color_mine_shaft_approx;
  opacity: 0.6;
  position: absolute;
  top: 0;
  left: 0;
  animation: sk-bounce 2.0s infinite ease-in-out;
}


.spinner {
  width: 150px;
  height: 150px;
  position: relative;
  margin: 100px auto;
}
.doubleBounce1 {
  @extend %extend_1;
}
.doubleBounce2 {
  @extend %extend_1;
  animation-delay: -1.0s;
}
@-webkit-keyframes sk-bounce {
  0% {
  }
  100% {
  }
  50% {
  }
}
@keyframes sk-bounce {
  0% {
    //Instead of the line below you could use @include transform($scale, $rotate, $transx, $transy, $skewx, $skewy, $originx, $originy)
    transform: scale(0.0);
  }
  100% {
    //Instead of the line below you could use @include transform($scale, $rotate, $transx, $transy, $skewx, $skewy, $originx, $originy)
    transform: scale(0.0);
  }
  50% {
    //Instead of the line below you could use @include transform($scale, $rotate, $transx, $transy, $skewx, $skewy, $originx, $originy)
    transform: scale(1.0);
  }
}