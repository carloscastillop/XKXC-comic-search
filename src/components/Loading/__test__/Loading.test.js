import React from 'react';
import {render} from '@testing-library/react';
import Loading from '../Loading';
import renderer from 'react-test-renderer';

test('Get Loading container', () => {
    const {getByTestId} = render(<Loading/>);
    const linkElement = getByTestId('loading');
    expect(linkElement).toBeInTheDocument();
});

test('Loading snapshop', () => {
    const tree = renderer
        .create(<Loading/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
