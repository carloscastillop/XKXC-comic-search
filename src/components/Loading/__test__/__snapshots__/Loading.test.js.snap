// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading snapshop 1`] = `
<div
  className="spinner"
  data-testid="loading"
>
  <div
    className="doubleBounce1"
  />
  <div
    className="doubleBounce2"
  />
</div>
`;
